File: android/app/src/main/java/com/mfsample/ActivityManager.java
package com.mfsample;

import android.app.Activity;

import java.util.ArrayList;
import java.util.List;

public class ActivityManager {
    private static ActivityManager instance;
    private List<Activity> activities;

    private ActivityManager() {
        activities = new ArrayList<>();
    }

    public static synchronized ActivityManager getInstance() {
        if (instance == null) {
            instance = new ActivityManager();
        }
        return instance;
    }

    public void addActivity(Activity activity) {
        activities.add(activity);
    }

    public void removeActivity(Activity activity) {
        activities.remove(activity);
    }

    public void finishActivity(Class<?> clazz) {
        for (Activity activity : activities) {
            if (activity.getClass().equals(clazz)) {
                activity.finish();
                break;
            }
        }
    }

    public boolean hasActivity(Class<?> clazz) {
        boolean isActivityAvailable = false;
        for (Activity activity : activities) {
            if (activity.getClass().equals(clazz)) {
                isActivityAvailable = true;
                break;
            }
        }
        return isActivityAvailable;
    }

}

File: android/app/src/main/java/com/mirrorfly_rn/ActivityManager.kt
package com.mirrorfly_rn

import android.app.Activity

class ActivityManager private constructor() {
    private val activities: MutableList<Activity> =
        ArrayList()

    fun addActivity(activity: Activity) {
        activities.add(activity)
    }

    fun removeActivity(activity: Activity) {
        activities.remove(activity)
    }

    fun finishActivity(clazz: Class<*>) {
        for (activity in activities) {
            if (activity.javaClass == clazz) {
                activity.finish()
                break
            }
        }
    }

    fun hasActivity(clazz: Class<*>): Boolean {
        var isActivityAvailable = false
        for (activity in activities) {
            if (activity.javaClass == clazz) {
                isActivityAvailable = true
                break
            }
        }
        return isActivityAvailable
    }

    companion object {
        @get:Synchronized
        var instance: ActivityManager? = null
            get() {
                if (field == null) {
                    field = ActivityManager()
                }
                return field
            }
            private set
    }
}
